### PR TITLE
add post & delete methods

### PR DESCRIPTION
**Pull request by AmineTir**

### CODE REVIEW

File: src/queries.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ export function deleteNodes(ids: number[]) {
     const stateNodeIndex = dbTable.findIndex((node) => node.id === head)
     dbTable.splice(stateNodeIndex, 1)
   }
+  return dbTable
```
Comment (atahrijouti) [inline src/queries.ts]: why return the entire array?

File: src/server.ts [hunk 2]
```diff
@@ -16,6 +16,16 @@ router.get("/nodes/:id", (ctx) => {
   ctx.body = getNodeAndChildren(Number(ctx.params.id))
 })
 
+router.post("/nodes", (ctx) => {
+  const body = ctx.request.query
+  ctx.body = storeNewNode(body.name, body.type, Number(body.parent))
+})
+
+router.del("/nodes/:ids", (ctx) => {
+  const ids = ctx.params.ids.split("-").map(Number)
```
Comment (atahrijouti) [inline src/server.ts]: can't the params be sent in the body of the request somehow instead?

File: src/server.ts [hunk 3]
```diff
@@ -16,6 +16,16 @@ router.get("/nodes/:id", (ctx) => {
...
+  ctx.body = deleteNodes(ids)
```
Comment (atahrijouti) [inline src/server.ts]: I would just return empty string and status to confirm that modifications have been made

File: src/server.ts (hunk 2 above)
Comment (AmineTir) [reply src/server.ts]: I did some research & couldn't make it work using `ctx.request.body`

Comment (atahrijouti) [reply src/server.ts]: https://github.com/koajs/bodyparser

Comment (AmineTir) [reply src/server.ts]: Done

File: src/server.ts [hunk 4]
```diff
@@ -16,11 +17,23 @@ router.get("/nodes/:id", (ctx) => {
...
 })
 
+router.post("/nodes", (ctx) => {
+  const body = ctx.request.body
+  ctx.body = storeNewNode(body.name, body.type, Number(body.parent))
+})
+
+router.del("/nodes", (ctx) => {
+  deleteNodes(ctx.request.body.ids)
+  ctx.body = "Deleted"
```
Comment (abderrahmane-tj) [inline src/server.ts:27]: status 204 means, "No Content". Maybe we don't need to return any text in the body?

File: src/server.ts [hunk 5]
```diff
@@ -16,11 +17,23 @@ router.get("/nodes/:id", (ctx) => {
   ctx.body = getNodeAndChildren(Number(ctx.params.id))
 })
 
+router.post("/nodes", (ctx) => {
+  const body = ctx.request.body
+  ctx.body = storeNewNode(body.name, body.type, Number(body.parent))
```
Comment (atahrijouti) [inline src/server.ts]:
```suggestion
  ctx.body = storeNewNode(body.name, body.type, body.parentId)
```

Comment (abderrahmane-tj) [reply src/server.ts:22]: Shit, how do you create a node inside the root? `Number(body.parentId)` seems to convert `null` into `0` :P

Comment (atahrijouti) [reply src/server.ts]: No need for type casting if you we are sending json body. a number is a number. if it's null it's null and that's that :P

File: src/server.ts (hunk 4 above)
Comment (atahrijouti) [inline src/server.ts]:
```suggestion
```

**Review by atahrijouti — APPROVED** [review]

**Pull request merged**